File: src/core/java/net/remmintan/mods/minefortress/core/interfaces/blueprints/IBlueprintMetadata.java
package net.remmintan.mods.minefortress.core.interfaces.blueprints;

import net.minecraft.util.BlockRotation;

public interface IBlueprintMetadata {
    String getName();

    String getId();

    String getRequirementId();

    int getFloorLevel();

    void setFloorLevel(int floorLevel);

    void rotateRight();

    void rotateLeft();

    BlockRotation getRotation();
}

File: src/core/java/net/remmintan/mods/minefortress/core/interfaces/blueprints/IBlueprintMetadata.kt
package net.remmintan.mods.minefortress.core.interfaces.blueprints;

import net.minecraft.util.BlockRotation;

public interface IBlueprintMetadata {
    String getName();

    String getId();

    IBlueprintRequirement getRequirement();

    int getFloorLevel();

    void setFloorLevel(int floorLevel);

    void rotateRight();

    void rotateLeft();

    BlockRotation getRotation();
}
